### PR TITLE
V1.5 unique ids for tasks

### PR DESCRIPTION
**Pull request by josheide**
Add an identifier field to the tasks. It can be taken from a counter (first task is 1, second is 2). The ID should be generated by the program and not the user. ✓
Modify the list command so it shows the IDs of the tasks ✓
Modify the add command so it always shows what ID was used for the new task. ✓
Modify the delete and complete commands so they take the ID of the task instead of its name.✓
Think about the users who update from the previous version of your program. Their stored file will be in the previous format. Make sure not to break anything for them and provide a nice migration process to this new version.✓

### CODE REVIEW

File: src/Main.java [hunk 1]
```diff
@@ -44,7 +41,7 @@ public static void main(String[] args) {
         System.out.println("");
 
         while (true) {
-            System.out.print("Enter a task (or 'exit' to save and exit): ");
+            System.out.print("Enter a command (or 'exit' to save and exit): ");
             String userInput = scanner.nextLine();
 
             if (userInput.equals("add")) {
```
Comment (balazskiss) [inline src/Main.java:47]: add command should not accept empty task name

File: src/Main.java [hunk 2]
```diff
@@ -56,8 +53,8 @@ public static void main(String[] args) {
                 String userAssigned = scanner.nextLine();
 
                 if (!userAssigned.isEmpty()) {
-                    taskList.add(new Task(userTask, false, userAssigned));
-                    System.out.println(userTask + " has been added!");
+                    taskList.add(new Task(nextID, userTask, false, userAssigned));
+                    System.out.println(userTask + " has been added with the unique identifier: " + taskList.size()); // If you delete a task then the ID that it prints here is off. I tried nextID here and it always prints 1.
```
Comment (balazskiss) [inline src/Main.java:57]: The output here seems incorrect. When I add a task it reports a unique ID but then in the `list` command output I see a different ID.
This is a bug.

File: src/Task.java [hunk 3]
```diff
@@ -1,15 +1,26 @@
...
+        return id;
+    }
 
-    Task(String name, boolean isComplete, String assignedToUser) {
-        this.name = name;
+    // This new getter method is used when deleting or completing a task with an ID.
 
-        this.isComplete = isComplete;
 
+    Task(int id, String name, boolean isComplete, String assignedToUser) {
```
Comment (balazskiss) [inline src/Task.java:16]: seems like the id parameter is unused here?

Comment (balazskiss) [reply src/Task.java:16]: if it's unused it should not be there.

File: src/Main.java [hunk 4]
```diff
@@ -19,6 +13,8 @@ public static void main(String[] args) {
 
         ArrayList<Task> taskList = new ArrayList<Task>();
 
+        int nextID = taskList.size() + 1;
```
Comment (balazskiss) [inline src/Main.java:16]: nextID is in two places now: 1) here and 2) in Task class as a static variable. Thee two values can get out of sync which leads to bugs in the program.
I recommend to simplify this and have a single source of truth for nextID.

File: src/Main.java [hunk 5]
```diff
@@ -2,15 +2,9 @@
 // then press Enter. You can now see whitespace characters in your code.
 import java.util.Scanner;
 import java.util.ArrayList;
-import java.io.File;
-import java.io.FileReader;
-import java.io.BufferedReader;
-import java.io.BufferedWriter;
-import java.io.FileWriter;
-import java.io.IOException;
+import java.io.*; // I replaced .File .FileReader .BufferedReader .BufferedWriter .FileWriter .IOException with *
```
Comment (balazskiss) [inline src/Main.java:5]: No need to leave this comment in the code.

**Review by balazskiss — CHANGES_REQUESTED** [review]
there are some bugs to be squashed :)

File: src/Main.java (hunk 2 above)
Comment (josheide) [reply src/Main.java:57]: This was a really tricky one to solve but I think I figured it out!

File: src/Main.java (hunk 1 above)
Comment (josheide) [reply src/Main.java:47]: Got it!

File: src/Task.java (hunk 3 above)
Comment (josheide) [reply src/Task.java:16]: Replaced it with nextId that I actually use.

File: src/Main.java (hunk 5 above)
Comment (josheide) [reply src/Main.java:5]: I just left it here in case this broke my code so that I would remember what to re-add :D

**Review by balazskiss — APPROVED** [review]

**Review by balazskiss — APPROVED** [review]

**Review by balazskiss — APPROVED** [review]

**Review by balazskiss — APPROVED** [review]

**Pull request merged**